### PR TITLE
US 8.5:  My Friends List

### LINKED ISSUE #278: US-8.5: Friends List
**User Story Description**

As a user, I want to be able to see my list of friends so that I can see who I am friends with.

**User Acceptance Flow**

1. User logs in
1. User goes to their Profile Page
1. User clicks on 'My Friends'
1. User is able to see a list of their friends in alphabetical order

**Acceptance Criteria**
- [ ] Friends query returns true friends (where you are both friends with each other)
- [ ] User Acceptance Flow is respected
- [ ] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [ ] Documentation present where necessary
- [ ] Unit and UI tests are implemented for every new component and function
- [ ] The application should still work with seeded data
- [ ] Must match mockups:
   - "[My Friends](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1588%3A4821)" view
   - "[More info modal](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1596%3A4543)" view

### PR DESCRIPTION
**Pull request by r-saba**
closes #278 

To test visit `/my-friends`. Ensure your user has bidirectional relationship with at least one other use in friends table.

To test the e2e, run the seeder first or ensure `user 1` has at at least 2 friends

### CODE REVIEW

File: apps/mull-ui/src/app/pages/profile/myFriends-profile/myFriends-profile.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      <div className="friend-category-container" key={key}>
+        <h2 className="friend-category-title">{key}</h2>
+        {currentCategoryContactRows}
+      </div>
+    );
+    finalFriendsList.push(currentLetterCategorizedList);
+  });
+
+  return (
+    <div className="page-container my-friends">
```
Comment (RGPosadas) [inline apps/mull-ui/src/app/pages/profile/myFriends-profile/myFriends-profile.tsx:82]: Blocking: Missing the [MullBackButton](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1588%3A4821) from the mockup

File: apps/mull-ui/src/app/pages/profile/myFriends-profile/myFriends-profile.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      <h1 className="my-friends-page-title">My Friends</h1>
+      {finalFriendsList}
```
Comment (RGPosadas) [inline apps/mull-ui/src/app/pages/profile/myFriends-profile/myFriends-profile.tsx]: We should tell the user that the page is empty when there are no friends
```suggestion
      {finalFriendsList.length > 0 ? finalFriendsList : 'No friends found.'}
```

**Review by RGPosadas — CHANGES_REQUESTED** [review]
There is also one linting error:
```js
pwd/apps/mull-ui-e2e/src/integration/US-8.5.spec.ts
20:11  error  'count' is never reassigned. Use 'const' instead  prefer-const
```

File: apps/mull-ui/src/app/pages/profile/myFriends-profile/myFriends-profile.tsx (hunk 2 above)
Comment (cristian-aldea) [reply apps/mull-ui/src/app/pages/profile/myFriends-profile/myFriends-profile.tsx]: rip users with no friends, we're just rubbing it in

**Review by cristian-aldea — APPROVED** [review]
- [x] Friends query returns true friends (where you are both friends with each other)
- [x] User Acceptance Flow is respected
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data
- [x] Must match mockups:
   - "[My Friends](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1588%3A4821)" view
   - "[More info modal](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1596%3A4543)" view

**Review by RGPosadas — APPROVED** [review]
LGTM~

**Acceptance Criteria**
- [x] Friends query returns true friends (where you are both friends with each other)
- [x] User Acceptance Flow is respected
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data
- [x] Must match mockups:
   - "[My Friends](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1588%3A4821)" view
   - "[More info modal](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=1596%3A4543)" view

**Pull request merged**